### PR TITLE
testutils.github: use sticky comments

### PR DESCRIPTION
**Pull request by miri64**
This makes Murdock less spammy in the release candidate test tracking issues.

See ~~https://github.com/RIOT-OS/Release-Specs/issues/227#issuecomment-874236990~~ https://github.com/RIOT-OS/Release-Specs/issues/227#issuecomment-874717209 for a currently generated example comment.

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Something isn't looking right for me, stdout is showing on 01. Task 03 - Unittests on native separated [tests-gcoap] and below... Could it be a stray escape character?

**Comment by miri64** [comment]
> Something isn't looking right for me, stdout is showing on 01. Task 03 - Unittests on native separated [tests-gcoap] and below... Could it be a stray escape character?

Yeah yeah, I tried to fix something different (the **SKIPPED** info output), and accidentally broke the other output in the process :sweat_smile:. I restarted the tests, the table should fix itself in the process.

**Comment by miri64** [comment]
(with the last force push both issues should be fixed)

**Comment by miri64** [comment]
> I restarted the tests, the table should fix itself in the process.

Done.

**Comment by miri64** [comment]
Damn, there is a character limit on comments (65536 characters). So now the comment has become too long and there are no more comments left. Will try to find a solution for that.

**Comment by miri64** [comment]
Task 4.1 alone apparently alone exceeds this character limit. I am a bit confused, how https://github.com/RIOT-OS/Release-Specs/issues/227#issuecomment-874040873 can be then, but I _do_ get an error message that I exceeded the character limit with this PR [here](https://github.com/RIOT-OS/Release-Specs/pull/228/files#diff-9d99c0b27cd7d383a7053ed1e03e8611ff67dafc95d69dad4a6d1b8d3c6de312R294).

**Comment by MrKevinWeiss** [comment]
I am not opposed to having a summary of x tests passed, y tests failed, and z tests skipped with a dropdown containing a list of each fail test and a link to an external log dump?  Do we trust the system enough to not need to look into if a test passes?

**Comment by miri64** [comment]
> I am not opposed to having a summary of x tests passed, y tests failed, and z tests skipped with a dropdown containing a list of each fail test and a link to an external log dump?

I was thinking about the same, however there are two caveats to that:

1. We can't get a live update, as we only can link to the run in the GitHub action. That one only delivers the most current line to the frontend, so older runs than when the page is loaded won't be shown.
2. We can't link to the exact location, as the GitHub action line is pretty much disjunct from the pytest output line (and I am not even sure pytest knows how many lines it outputted so far)

So basically we can just provide a link to the action (as we already do in the spammy comments) and one needs to search for the corresponding output themselves when the task is done (or use the artifact HTML for less searching). If we are fine with that, I would prefer to remove the dumps altogether (even for the error cases) as even when just dumping the error cases we run into the risk of hitting the character limit (case and point Task 4.1 which has >80000 characters alone).

**Comment by miri64** [comment]
> Do we trust the system enough to not need to look into if a test passes?

I'd say yes. In the past the false negatives where more prevalent than the false positives.

**Comment by miri64** [comment]
Another (more complicated) alternative could be to dump the output to a private [gist](https://gist.github.com/) and link to that. But I would prefer to do that as a follow-up.

**Comment by miri64** [comment]
[Here](https://github.com/RIOT-OS/Release-Specs/issues/227#issuecomment-874717209) is how it would look like. Since of course there is no log of my tests online (another disadvantage), I added the required environment variables to link to the last release build just to show how it would look like when Murdock reports its results. For a local test, the outcome would just not be a link. I think the Gist variant [I proposed](https://github.com/RIOT-OS/Release-Specs/pull/228#issuecomment-874622224) could be a good solution for that as well.

**Comment by MrKevinWeiss** [comment]
> case and point Task 4.1 which has >80000 characters in its output alone

I wonder how much of that is usable information...

**Comment by MrKevinWeiss** [comment]
gist is nice, and somehow all in the github framework so image it would make implementation not horrible.

**Comment by miri64** [comment]
> gist is nice, and somehow all in the github framework so image it would make implementation not horrible.

Yepp, but still would prefer to do that as a follow up, since new tests would be needed to keep the coverage threshold up and I believe that would make this PR grow out of proportion.

**Comment by MrKevinWeiss** [comment]
Sure, I will need some time to look through everything.

**Comment by MrKevinWeiss** [comment]
The code looks ok, ya 400+ lines are a lot but much of that is tests and writing parsers out step by step (which I prefer to one-line regex magic).

It is an improvement and I wouldn't mind trying it out for RC-2.  Is there anything else you wanted to add for this one?  (if not please squash)

**Comment by miri64** [comment]
Squashed. The Gist output is in the works. But as I said: I would prefer to provide it as a follow-up.

**Review by MrKevinWeiss — APPROVED** [review]
Looks good, I think the follow up with Gist would be pretty nice as looking through the failures is a bit of a challenge now.

ACK!

**Comment by MrKevinWeiss** [comment]
Just thinking out loud but it would also be cool to have some easy to diff output to show a comparison of tests that have failed from the last releases (as experimental sometimes means expected failure).

**Comment by miri64** [comment]
> Just thinking out loud but it would also be cool to have some easy to diff output to show a comparison of tests that have failed from the last releases (as experimental sometimes means expected failure).

Mhhh this could be achieved by having a results repo instead of a gist I guess and then marking the latest build with a tag named like the release (candidate). Of course, gists are Git repos too, but since they are only identified by a hash it is quite hard to find them again when not stored somewhere (which in the current case I do with an HTML comment in the sticky comment). Maybe another future improvement. However, that would require touching the `release-test.yaml` in RIOT-OS/RIOT though as the state needs to come from the environment.
